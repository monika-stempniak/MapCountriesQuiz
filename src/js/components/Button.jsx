// @flow
import * as React from 'react'

type Props = {
  htmlType?: string,
  btnClass: string,
  isHidden?: boolean,
  handleClick?: (e: SyntheticMouseEvent<HTMLAnchorElement>) => void,
  children: React.Node,
}

const Button = (props: Props) => {

  const {
    htmlType,
    btnClass,
    isHidden,
    handleClick,
    children,
  } = props

  return (
    <button
      type={htmlType}
      className={btnClass}
      aria-hidden={isHidden}
      onClick={handleClick}
    >
      {children}
    </button>
  )
}

Button.defaultProps = {
  htmlType: 'button',
  isHidden: false,
  handleClick: () => {},
};

export default Button
