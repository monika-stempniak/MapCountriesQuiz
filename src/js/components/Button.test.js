import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('<Button />', () => {
  const wrapper = shallow(<Button />)
  it('should render a button element', () => {
    expect(wrapper.find('button[type="button"]').exists()).toBeTruthy()
    expect(wrapper.prop('isHidden')).toBeFalsy()
  })
  it('matches the snapshot', () => {
    const tree = shallow(<Button />)
    expect(tree).toMatchSnapshot()
  })
})
