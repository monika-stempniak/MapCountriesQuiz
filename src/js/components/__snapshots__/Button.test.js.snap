// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches the snapshot 1`] = `
<button
  aria-hidden={false}
  type="button"
/>
`;
